kaggle_environments/helpers.py

```diff
@@ -172,3 +147,59 @@ def act_timeout(self) -> float:
     def run_timeout(self) -> float:
         """Maximum runtime (seconds) of an episode (not necessarily DONE)."""
         return self["runTimeout"]
+
+
+TConfiguration = TypeVar('TConfiguration', bound=Configuration)
+TObservation = TypeVar('TObservation', bound=Observation)
+TAction = TypeVar('TAction')
+Agent = Callable[[TObservation, TConfiguration], TAction]
+
+
+class AgentStatus(Enum):
+    UNKNOWN = 0
+    ACTIVE = 1
+    INACTIVE = 2
+    DONE = 3
+    INVALID = 4
+    ERROR = 5
+
+
+class AgentState(Generic[TObservation, TAction], Dict[str, any]):
```

Comment:
These new classes are not used yet, please ignore them for now and they'll be employed as part of a larger refactor.